### ISSUE TITLE
Can't use module outputs as input to another module

### ISSUE THREAD

**Issue by mcoms** [hashicorp/terraform#822]
If I try to use a module output as the input to another module, I get an error when running `terraform plan`, v0.3.5.

```
# main.tf
module "cluster_leader" {
  source = "./leader"
}

module "cluster_follower_1" {
  source = "./follower"
  leader_ip = "${module.cluster_leader.private_ip}"
}

module "cluster_follower_2" {
  source = "./follower"
  leader_ip = "${module.cluster_leader.private_ip}"
}
```

```
# leader/main.tf
...
output "private_ip" { value = "${aws_instance.server.private_ip}" }
```

```
# follower/main.tf
variable "leader_ip" {}
...
```

```
➜  terraform plan
Refreshing Terraform state prior to plan...
...
Error refreshing state: Output field 'private_ip' not found for variable 'module.cluster_leader.private_ip'
```

Ideally, I'd like for Terraform to calculate that the followers depend on the leader, or the option to specify `depends_on` for modules.

**Comment by mitchellh** [comment]
I believe this was fixed in 0.3.6. But I've tagged it as a bug and will verify later.

**Comment by xueshanf** [comment]
I am on 0.3.6 version and just tried it,  same problem. Originally reported the issue on version 0.3.5, #697.

**Comment by mitchellh** [comment]
This is fixed on master, since #1010

**Comment by mcoms** [comment]
Thanks! :+1:
